### DISCUSSION TITLE
Rewrites adding unnecessary path query parameters

### DISCUSSION THREAD

**Discussion by joakimgunst** [vercel/next.js#16984] [Help] [answered]
When adding a rewrite like:

```js
async rewrites() {
  return [
    {
      source: "/api/:path*",
      destination: "http://localhost:8080/api/:path*",
    }
  ];
},
```

The logs will show requests like `http://localhost:8080/api/v1/products?path=v1&path=products` with unnecessary path query parameters, which Next.js seems to add automatically. This is not directly harmful, but will make the logs harder to read. Is there any way to disable this? The [docs](https://nextjs.org/docs/api-reference/next.config.js/rewrites) don't seem to mention it.

**Comment by ijjk** [comment, marked as answer]
Hi, this has been updated since `v9.5.3` of Next.js to not add the additional query parameters if they are already used in the destination's path.

**Reply by joakimgunst** [reply]
Great! And thank you for the work with Next.js, it's a fantastic framework.
